test/NJsonApiCore.Test/Serialization/JsonApiTransformerTest/TestMetaData.cs

```diff
@@ -0,0 +1,206 @@
+﻿using NJsonApi.Infrastructure;
+using NJsonApi.Serialization.Documents;
+using NJsonApi.Serialization.Representations.Resources;
+using NJsonApi.Test.Builders;
+using NJsonApi.Test.TestControllers;
+using System;
+using System.Collections.Generic;
+using Xunit;
+using Newtonsoft.Json;
+using NJsonApi.Serialization.Representations.Relationships;
+using NJsonApi.Serialization.Representations;
+
+namespace NJsonApi.Test.Serialization.JsonApiTransformerTest
+{
+    public class TestMetaData
+    {
+        [Fact]
+        public void Creates_CompondDocument_for_single_class_with_nometadata_and_propertly_map_nometadata()
+        {
+            // Arrange
+            var context = CreateContext();
+            SampleClass objectToTransform = CreateObjectToTransform();
+            var transformer = new JsonApiTransformerBuilder()
+                .With(CreateConfiguration())
+                .Build();
+
+            // Act
+            CompoundDocument result = transformer.Transform(objectToTransform, context);
+
+            // Assert
+            var transformedObject = result.Data as SingleResource;
+            Assert.Null(transformedObject.MetaData);
+        }
+
+        [Fact]
+        public void Creates_CompondDocument_for_single_class_with_metadata_and_propertly_map_metadata()
+        {
+            // Arrange
+            var context = CreateContext();
+            SampleClassWithObjectMetadata objectToTransform = CreateObjectWithMetadataToTransform();
+            var transformer = new JsonApiTransformerBuilder()
+                .With(CreateConfiguration())
+                .Build();
+
+            // Act
+            CompoundDocument result = transformer.Transform(objectToTransform, context);
+
+            // Assert
+            var transformedObject = result.Data as SingleResource;
+            Assert.Equal("value1", transformedObject.MetaData["meta1"]);
+            Assert.Equal("value2", transformedObject.MetaData["meta2"]);
+        }
+
+        [Fact]
+        public void Creates_CompondDocument_for_single_class_with_relationship_metadata_and_propertly_map_metadata()
+        {
+            // Arrange
+            var context = CreateContext();
+            SampleClassWithRelatedClassesWithRelationshipMetadata objectToTransform = CreateObjectWithRelationshipMetadataToTransform();
+            var transformer = new JsonApiTransformerBuilder()
+                .With(CreateConfiguration())
+                .Build();
+
+            // Act
+            CompoundDocument result = transformer.Transform(objectToTransform, context);
+
+            // Assert
+            var transformedObject = result.Data as SingleResource;
+            Assert.Equal("value100", ((SingleResourceIdentifier)(transformedObject.Relationships["relatedObject"]).Data).MetaData["meta100"]);
+            Assert.Equal("value1", ((MultipleResourceIdentifiers)(transformedObject.Relationships["relatedObjects"]).Data)[0].MetaData["meta1"]);
```

Comment:
Worth adding a second element to the array with it's own relationship metadata just to ensure we don't get the same relationship metadata for all members of the array?